### PR TITLE
Variable Authentication: Allow users to authenticate into Jenkins with Hubot

### PR DESCRIPTION
**Pull request by farmdawgnation**
This PR creates the ability for a user to provide their own Jenkins authentication information to yardmaster so actions they request will be executed using their Jenkins account.

To authenticate you can do:

```
> hubot jenkins auth set farmdawgnation XXXXXXXXX
Attempting to authenticate you as farmdawgnation
Done. From now on I'll authenticate your requests to jenkins as farmdawgnation.
```

Additionally you can check your current authentication:

```
> hubot jenkins auth
You are authenticated with Jenkins as farmdawgnation
```

And clear your authentication

```
> hubot jenkins auth clear
I've removed your Jenkins credentials. Commands you issue will use my Jenkins credentials.
```

### CODE REVIEW

File: src/yardmaster.coffee [hunk 1]
```diff
@@ -508,6 +535,9 @@ module.exports = (robot) ->
     )
     cronjob.start()
 
+  robot.respond /yardmaster auth ([^ ]+) (.+)/i, (msg) ->
```
Comment (riveramj) [inline src/yardmaster.coffee]: thoughts on `[yardmaster|jenkins]`? since most people will know it as a jenkins interaction

**Comment by riveramj** [comment]
if you enter in an incorrect api key (like missed the last letter), it'll still say `done` but when you try to build, it wont respond right away and eventually return
>ERROR SyntaxError: Unexpected token <
  at Object.parse (native)

is there any way we can check if the auth is correct and return a message as appropriate?

**Comment by farmdawgnation** [comment]
Yeah, let me see what I can do about that.

File: src/yardmaster.coffee (hunk 1 above)
Comment (farmdawgnation) [inline src/yardmaster.coffee]: Jenkins probably makes more sense.

**Comment by farmdawgnation** [comment]
Alright, gave this another pass with some added functionality to boot. Let me know what you think.

File: src/yardmaster.coffee [hunk 2]
```diff
@@ -508,6 +571,15 @@ module.exports = (robot) ->
     )
     cronjob.start()
 
+  robot.respond /jenkins auth/i, (msg) ->
```
Comment (riveramj) [inline src/yardmaster.coffee]: need an end of string anchor here. If you run `clear`, this will match as well.

**Comment by riveramj** [comment]
Everything else tests :100:.

I looked at merge conflict and I think its just that `buildBranch` and `buildFeedback` where in the same spot but gonna let you fix that, as it should only take you a second.

will merge after that!

**Comment by farmdawgnation** [comment]
:white_check_mark:

**Comment by farmdawgnation** [comment]
Imma bout to turbo merge this if ya don't get to it soon Mike :P

**Comment by riveramj** [comment]
:tada:

**Pull request merged**